polyfill/lib/ecmascript.mjs

```diff
@@ -377,58 +377,34 @@ export const ES = ObjectAssign({}, ES2019, {
     nanoseconds,
     overflow
   ) => {
+    for (const prop of [years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds]) {
+      if (!Number.isFinite(prop)) throw new RangeError('infinite values not allowed as duration fields');
+    }
     ES.RejectDurationSign(years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds);
-    switch (overflow) {
-      case 'reject':
-        for (const prop of [
-          years,
-          months,
-          weeks,
-          days,
-          hours,
-          minutes,
-          seconds,
-          milliseconds,
-          microseconds,
-          nanoseconds
-        ]) {
-          if (!Number.isFinite(prop)) throw new RangeError('infinite values not allowed as duration fields');
-        }
-        break;
-      case 'constrain': {
-        const arr = [years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds];
-        for (const idx in arr) {
-          if (!Number.isFinite(arr[idx])) arr[idx] = Math.sign(arr[idx]) * Number.MAX_VALUE;
-        }
-        [years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds] = arr;
-        break;
-      }
-      case 'balance': {
-        ({ days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.BalanceDuration(
-          days,
-          hours,
-          minutes,
-          seconds,
-          milliseconds,
-          microseconds,
-          nanoseconds,
-          'days'
-        ));
-        for (const prop of [
-          years,
-          months,
-          weeks,
-          days,
-          hours,
-          minutes,
-          seconds,
-          milliseconds,
-          microseconds,
-          nanoseconds
-        ]) {
-          if (!Number.isFinite(prop)) throw new RangeError('infinite values not allowed as duration fields');
-        }
-        break;
+    if (overflow === 'balance') {
+      ({ days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.BalanceDuration(
+        days,
+        hours,
+        minutes,
+        seconds,
+        milliseconds,
+        microseconds,
+        nanoseconds,
+        'days'
+      ));
+      for (const prop of [
+        years,
+        months,
+        weeks,
+        days,
+        hours,
+        minutes,
+        seconds,
+        milliseconds,
+        microseconds,
+        nanoseconds
+      ]) {
+        if (!Number.isFinite(prop)) throw new RangeError('infinite values not allowed as duration fields');
```

Comment:
(Feel free to file an issue on this instead.) Is it possible for `BalanceDuration` to return an invalid result like this? Should it be? Do other call sites check / need to check for it?

Reply:
Yeah, it's a good question ... I wondered it while I was writing this code, but forgot about it.

Reply:
Yes, this is possible: `Temporal.Duration.from({ seconds: Number.MAX_VALUE, milliseconds: Number.MAX_VALUE }, { overflow: 'balance' })`

I'll open a new issue for it, since I discovered a separate bug while testing this.

Reply:
→ #915